### PR TITLE
CI: Various minor improvements to fix CI

### LINKED ISSUE #185: CI tests are failing
- Failure at https://github.com/computationalmodelling/nbval/runs/8236634175?check_suite_focus=true#step:5:80

- became known while running CI for merge request #184, which - however - only changes the Readme of the project (https://github.com/computationalmodelling/nbval/pull/184/files).

### PR DESCRIPTION
**Pull request by bsipocz**
This should work around one of the CI issues. I can't yet reproduce the other one locally.

fixes #185

### CODE REVIEW

**Comment by bsipocz** [comment]
I suppose looking at other PRs would have been a quicker way to figure out the mpl version changes, nevertheless, I feel the addition of some verbosity for the versions is still useful, so I left all the debug commits in there.

Also, this now includes the cherry-picked commit from https://github.com/computationalmodelling/nbval/pull/183 as a way to trigger CI for that one.

**Comment by bsipocz** [comment]
cc @fangohr  - this is ready to go. Let me know if any of the changes are controversial and if you prefer them to be removed.

File: tests/sanitize_defaults.cfg [hunk 1]
```diff
@@ -13,4 +13,8 @@ replace: TIMESTAMP
 
 [Memory addresses]
 regex: (<[a-zA-Z_][0-9a-zA-Z_.]* at )(0x[0-9a-fA-F]+)(>)
-replace: \1MEMORY_ADDRESS\3
\ No newline at end of file
+replace: \1MEMORY_ADDRESS\3
+
+[Matplotlib figure size]
+regex: (Figure size )\d+x\d+( with \d+ Axes)
+replace: \1WIDTHxHEIGHT\2
```
Comment (fangohr) [inline tests/sanitize_defaults.cfg:20]: @bnavigator @bsipocz what is the purpose of `\1` and `\2` at the beginning and end of the replacement string (if you know)?

(And why is it `\2` at the end of the `WIDTHxHEIGHT` string but `\3` at the end of the `MEMORY_ADDRESS` string?)

**Review by fangohr — COMMENTED** [review]
This looks great @bsipocz, thank you for that contribution! I like the additional `pip freeze` debug info, moving to python 3.10, and sanitising the figure size.

Thank you @bnavigator for the sanitizing figure contribution in #183 !

I left one minor query in the code changes.

File: tests/sanitize_defaults.cfg (hunk 1 above)
Comment (bsipocz) [reply tests/sanitize_defaults.cfg:20]: those are references for the regex groups. I haven't looked into the logic, but I expect that the aim is to ignore only the smallest, unpredictable part of the output while still failing the tests if e.g a figure with more Axes is generated than expected one.

As for groups, there are 3 of them in the memory addresses case and 2 in this one. I suppose the middle part, which is to be ignored, can be put in a group, but it shouldn't really matter in this case as there are no plans to reuse that group.

Comment (fangohr) [reply tests/sanitize_defaults.cfg:20]: Of course; very clever. Thank you!

**Pull request merged**